src/python/pants/backend/python/rules/run_setup_py_test.py

```diff
@@ -54,3 +54,150 @@ def alias_groups(cls):
 
   def tgt(self, addr: str) -> HydratedTarget:
     return self.request_single_product(HydratedTarget, Params(Address.parse(addr)))
+
+
+class TestGetOwnedDependencies(TestSetupPyBase):
+  @classmethod
+  def rules(cls):
+    return super().rules() + [
+      get_owned_dependencies,
+      get_exporting_owner,
+      RootRule(DependencyOwner),
+    ]
+
+  def assert_owned(self, owned: Iterable[str], exported: str):
+    assert sorted(owned) == sorted(
+      od.hydrated_target.address.reference() for od in self.request_single_product(
+      OwnedDependencies, Params(DependencyOwner(ExportedTarget(self.tgt(exported))))
+    ))
+
+  def test_owned_dependencies(self):
```

Comment:
Nit: adding `-> None` to all these tests will ensure MyPy checks the implementation of the functions.

Reply:
Done. Why wouldn't MyPy otherwise check them?

Reply:
Incremental type hinting. MyPy doesn't check function bodies unless you tell it to by adding at least one type hint.